### ISSUE TITLE
Adding documentation for Rust/Cargo extensions

### ISSUE THREAD

**Issue by pawandubey** [rubygems/rubygems#5648]
Hello,

The support for building Rust extensions added in https://github.com/rubygems/rubygems/pull/5175 is great! But we are lacking documentation on the proper way to use it. https://guides.rubygems.org/gems-with-extensions/ only talks about C extensions as well.

After trying to publish a gem with a Rust extension built with rutie, it seems like the ruby library has to require the `<gemname.so>` file for the integration to work. However, it is not clear how to build that shared library while developing the gem locally. Cargo builds usually end up in the `target` directory, and it only builds the `lib<gemname>.so` object (which incidentally ends up at `/path/to/ruby/extensions/<platform>/release`. through the Cargo extension builder).

Moreover, it seems like it expects us to check in the `Cargo.lock` file, whereas Cargo by default adds to to `.gitignore` for `lib` projects.

Basically, I think we can benefit from docs on how to structure the project correctly to be installed and worked on locally. I haven't yet figured out a way to make both of those possible at the same time, so while I am willing to contribute docs, I would appreciate pointers!

----
I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by deivid-rodriguez** [comment]
I think @ianks will most likely have some answers here :)

**Comment by ianks** [comment]
I recommend using the [rb_sys](https://github.com/oxidize-rb/rb-sys/tree/main/gem) for developing a gem locally. It handles all of that stuff, and more. As far as starting a new gem, #5613 will make getting started a lot easier.

I've started a bit on writing docs for it [in this Cargo book](https://oxidize-rb.github.io/rb-sys/). Although I consider Rust in Ruby to have feature parity with C at this point, the documentation is the biggest weakness at this point. I would be incredibly grateful for help on this!

If you are interested in helping, I would love to help you through this. Can you join [the oxidize-rb Slack](https://join.slack.com/t/oxidize-rb/shared_invite/zt-16zv5tqte-Vi7WfzxCesdo2TqF_RYBCw) to discuss more? @pawandubey

**Comment by pawandubey** [comment]
Thanks @deivid-rodriguez and @ianks! I will check out the links. Looks like things are already moving in the right direction, which is exciting! I will join the Slack and connect there. I am happy to leave this issue open for now, and close it once we have some documentation (even links to resources) rolled into the Rubygems guides, but no objection to closing it either.
